src/MARIGOLD/iate.py

```diff
@@ -0,0 +1,364 @@
+from .config import *
+
+def iate(cond, query, z_step = 0.01, dpdz_method = 'LM', void_method = 'driftflux', 
+         LM_C = 40, k_m = 0.40, LoverD_restriction = 9999,      # Pressure drop calculation arguments
+         cheat = True, elbow = False, quarantine = True):       # Temporary arguments, fix later
+    """
+    Version History:
+        > v1: Pressure cheating, jgref substitute for jgatm
+        > v2: MG update, pressure retrieval, jgatm retrieval
+        > v3: Yadav methods, incorporation of COV terms, support for elbows, VU, VD, horizontal
+    
+    Inputs:
+        > cond:             Condition object, part of MARIGOLD framework
+        > query:            L/D endpoint
+        > z_step:           Axial mesh cell size [-]
+        > void_method:      Void fraction prediction method, 'driftflux' or 'continuity'
+
+    Notes:
+        > IATE coefficients are currently set to default values depending on geometry
+            > Probably want to make these all optional arguments, set default values for 90 straight pipe, and input other values for different geometries outside of IATE function
+            > Same goes for COV models?
+    """
+
+    # MARIGOLD retrieval
+    theta           = cond.theta                                # Pipe inclination angle
+    Dh              = cond.Dh                                   # Hydraulic diameter
+    LoverD          = cond.LoverD                               # Condition L/D
+
+    # Yadav
+    RoverD = 9      # Make into optional input
+    R_curv          = RoverD * Dh                               # Radius of curvature of elbow/bend
+
+    # May want to include arc of the bend?
+    # Ah, yes
+    L_elb = np.pi/2 * R_curv        # This assumes 90\degree elbow
+    # So, Yadav defines under Expcond.m the length of the vertical section -- I think this is how he toggles between VU, H, VD, elbow
+    # Instead, we probably would want to have an input switch indicating which mode of IATE we're executing
+
+    ############################################################################################################################
+    #                                                                                                                          #
+    #                                                       CONSTANTS                                                          #
+    #                                                                                                                          #
+    ############################################################################################################################
+    # expcond.m
+
+    p_atm           = 101325                                    # Ambient pressure [Pa]
```

Comment:
Let's just put the properties in the Condition __init__, optional argument for phases. Then you can change all these to cond.value

Reply:
Resolved